File: investing-backend/src/main/java/com/davidmuns/investing/dto/PositionResponse.java
package com.davidmuns.investing.dto;

import java.time.LocalDate;

public record PositionResponse(
        Long id,
        String name,
        String symbol,
        String type,
        Double quantity,
        Long portfolioId,
        Double price,
        Double close,
        Double previousClose,
        Double fee,
        LocalDate createdAt,
        Double netAmount,
        Double grossAmount
) {
    public PositionResponse(Long id, String symbol, Double quantity, Long portfolioId, Double price, Double fee, LocalDate createdAt) {
        this(id, null, symbol, null, quantity, portfolioId, price, null, null, fee, createdAt, null, null);
    }
}

File: investing-frontend/src/app/shared/models/position-response.ts
export interface PositionResponse {
  id: number;
  name: string;
  symbol: string;
  type: string;
  fee: number;
  price: number;
  createdAt: string;
  quantity: number;
  portfolioId: number;
  close: number;
  previousClose: number;
  netAmount: number;
  grossAmount: number;
}
